include/gnc/utilities.hpp

```diff
@@ -15,60 +20,93 @@ namespace gnc {
-    lin::Vector<T, 4> const &q2, lin::Vector<T, 4> &res);
+        lin::Vector<T, 4> const &q2, lin::Vector<T, 4> &res);
 
 /** @fn rotate_frame
- *  Rotates the vector v according to the quaternion q. The result is stored in
- *  res. */
+ *  @param[in]  q   Quaternion specifying the frame rotation.
+ *  @param[in]  v   Vector to be transformed.
+ *  @param[out] res Result of the transformation.
+ *  Rotates a vector as specified by a quaternion. There is no explicit handling
+ *  of NaNs built into this function; however, a finite input will always yield
+ *  a finite result. */
 template <typename T>
-constexpr void rotate_frame(lin::Vector<T, 4> const &q, lin::Vector<T, 3> const &v,
-    lin::Vector<T, 3> &res);
+constexpr void rotate_frame(lin::Vector<T, 4> const &q,
+    lin::Vector<T, 3> const &v, lin::Vector<T, 3> &res);
 
 /** @fn rotate_frame
- *  Rotates the vector v in-place according to the quaternion q. */
+ *  @param[in]    q Quaternion specifying the frame rotation.
+ *  @param[inout] v Vector rotated in place. 
+ *  Rotates a vector as specified by a quaternion in place. There is no explicit
+ *  handling of NaNs built into this function; however, a finite input will
+ *  always yield a finite result. */
 template <typename T>
 constexpr void rotate_frame(lin::Vector<T, 4> const &q, lin::Vector<T, 3> &v);
 
 /** @fn dcm_to_quat
- *  Calculates the quaternion corresponding to the DCM m and stores the result
- *  in q. */ 
+ *  @param[in]  M Direction cosine matrix.
+ *  @param[out] q Output quaternion.
+ *  Determines the quaternion corresponding to the provided direction cosine
+ *  matrix. The input direction cosine matrix needs to be orthonormal. There is
+ *  no explicit handling of NaNs built into this function; however, a finite
+ *  input will always yield a finite result. */ 
 template <typename T>
-inline void dcm_to_quat(lin::Matrix<T, 3, 3> const &m, lin::Vector<T, 4> &q);
+inline void dcm_to_quat(lin::Matrix<T, 3, 3> const &M, lin::Vector<T, 4> &q);
 
 /** @fn triad
- *  Given a set of vectors in a knows frame (N_*), and a matching set of
- *  measured vectors in an unknown frame (B_*), this function will determine the
- *  quaternion required to rotate the unknown frame to the known frame. This
- *  result is stored in q.
- *  @returns Zero if the function was succesful and false otherwise.
- * 
- *  REQUIRES: N_* and B_* vectors must be unit vectors. */
+ *  @param[in]  R1 First known vector in a known frame.
+ *  @param[in]  R2 Second known vector in a known frame.
+ *  @param[in]  r1 First measurement vector in an unknown frame.
+ *  @param[in]  r2 Second measurement vector in an unknown frame.
+ *  @param[out] q  Quaternion transforming from the unknown to known frame.
+ *  Given a set of unit vector from a known frame and then unknown frame, this
+ *  function calculates the rotation required to transform from the latter to
+ *  the former. If R1 and R2, or r1 and r2 are within a degree of one another,
```

Comment:
I would add that q will rotate the first vector exactly, and the second vector as best it can.

Reply:
Oh shoot yep - will do

Reply:
Also that q is quat_unknown_known so it should rotate from known frame to unknown frame. Also, there are no MATLAB tests for triad, its copied from Justin's Simulink, and the docs in `utl_triad.m` aren't very helpful. #167

Reply:
Oh wait really? Welp, I may need to rework some documentation in other places as well. Really not sure where I got that from.

Reply:
And greatly improved the documentation in CXX at least for triad.